### PR TITLE
Improve fuzzing of both closed and open world styles of modules

### PR DESCRIPTION
**Pull request by kripken**
Before, we would simply not export a function that had an e.g. anyref
param. As a result, the modules were effectively "closed", which was
good for testing full closed-world mode, but not for testing degrees of
open world. To improve that, this PR allows the fuzzer to export such
functions, and an "enclose world" pass is added that "closes" the wasm
(makes it more compatible with closed-world) that is run 50% of the
time, giving us coverage of both styles.

### CODE REVIEW

File: src/passes/EncloseWorld.cpp [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+  // Make an enclosed stub function for an exported function, and return its
+  // name.
+  Name makeStubStubForExport(Function* func, Module* module) {
+    // Pick a valid name for the stub we are about to create.
+    auto stubName = Names::getValidFunctionName(
+      *module, std::string("stub$") + func->name.toString());
+
+    // Create the stub.
+    Builder builder(*module);
+    auto* stub = new Function();
```
Comment (tlively) [inline src/passes/EncloseWorld.cpp:101]:
```suggestion
    auto stub = std::make_unique<Function>();
```

File: src/passes/EncloseWorld.cpp [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    stub->name = stubName;
+    stub->hasExplicitName = true;
+
+    // The stub's body is just a call to the original function, but with some
+    // conversions to/from externref.
+    auto* call = module->allocator.alloc<Call>();
```
Comment (tlively) [inline src/passes/EncloseWorld.cpp:107]: It would be nice to use `Builder` (or even `IRBuilder` 👀) rather than manually allocating expressions.

**Review by tlively — APPROVED** [review]
Even if we close the `externref` workaround at some point, by then we will hopefully have the `@private` type annotations figured out, so this pass could apply those instead.

File: src/passes/EncloseWorld.cpp (hunk 2 above)
Comment (kripken) [reply src/passes/EncloseWorld.cpp]: Oh, sorry, I seem to have copied this old code and not noticed how old it was... Fixed.

**Review by tlively — APPROVED** [review]

**Pull request merged**